### DISCUSSION TITLE
Unable to login via iOS in version 1.26.0

### DISCUSSION THREAD

**Discussion by Buddinski88** [dani-garcia/vaultwarden#3008] [Q&A]
### Subject of the issue

I have updated my instance of Vaultwarden to version 1.26.0. Since then, I can no longer log into my vault with the latest version of the Bitwarden iOS app. Via browser or Windows app it works fine with the same password.

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.26.0
* Web-vault version: v2022.10.0
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: MySQL
* Database version: 8.0.29
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** SIGNUPS_ALLOWED, ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "*****://***********:***************************@***.***.***.*:****/***********",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://***********.**********.**",
  "domain_origin": "*****://***********.**********.**",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": false,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "****@***************.**",
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

* vaultwarden version: 1.26.0

* Installed via docker-compose with latest tag
  *  Docker version 20.10.21, build baeda1f
  *  Docker-compose version 1.29.2, build 5becea4c

* Clients used: Bittwarden Passwordmanager (iOS) 2022.11.0

* Reverse proxy and version:  Traefik 2.9.1

* MySQL version:  8.0.29

* Other relevant details:

### Steps to reproduce

* Perform standard installation via docker-compose with the latest tag
* Register user and try to login via the latest iOS app

### What i have did

First try:
* Switch back to version 1.25.2 > doesn't solve the issue

Second try:
* Switch `testing `version > doesn't solve the issue

Third try:
* Backup all data via json
* Delete local data volume and MySQL database
* Perform a clean installation
* Register new user and import the backup

### Expected behaviour

* Successful login as via the desktop application via the browser extension.

### Actual behaviour

* Login not possible via iOS app

### Troubleshooting data

```
[2022-12-19 08:49:17.804][request][INFO] POST /identity/connect/token
[2022-12-19 08:49:17.854][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: 172.70.250.128. Username: ****@***************.**.
[2022-12-19 08:49:17.854][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
[2022-12-19 08:49:43.069][request][INFO] POST /api/accounts/prelogin
[2022-12-19 08:49:43.070][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
[2022-12-19 08:49:43.200][request][INFO] POST /identity/connect/token
[2022-12-19 08:49:43.275][vaultwarden::api::identity][INFO] User ****@***************.** logged in successfully. IP: 172.68.110.114
[2022-12-19 08:49:43.275][response][INFO] (login) POST /identity/connect/token => 200 OK
[2022-12-19 08:49:43.550][request][INFO] POST /identity/connect/token
[2022-12-19 08:49:43.576][response][INFO] (login) POST /identity/connect/token => 200 OK
[2022-12-19 08:49:44.053][request][INFO] GET /api/sync
[2022-12-19 08:49:44.089][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
```
![IMG_5270](https://user-images.githubusercontent.com/50332516/208386198-b604d04b-f931-41cf-8605-084cae01f8c4.PNG)
![IMG_5271](https://user-images.githubusercontent.com/50332516/208386222-5b7b380c-46ba-42f2-8cb1-9a832ec05530.PNG)

**Comment by juanluisbujeda** [comment]
The same with de Safari extension and the app on OSX

**Comment by BlackDex** [comment]
have you tried the `testing` tagged version already?
The latest clients have some changes which could cause some issues which we already fixed in the `testing`/`main` branch

Also, i do see a successful login in the logs though. So, it seems to be possible to login.

**Comment by Buddinski88** [comment]
> have you tried the `testing` tagged version already? The latest clients have some changes which could cause some issues which we already fixed in the `testing`/`main` branch
> 
> Also, i do see a successful login in the logs though. So, it seems to be possible to login.

Yes, I have also made an attempt with the `testing`. Unfortunately also not successful.
The successful login has been via the Google Chrome browser extension.
The login via the iOS app is still not possible.

**Comment by BlackDex** [comment]
I doubt that it is a Vaultwarden issue though.
Double check your domain settings in the iOS app if that is correct and uses https.

**Comment by BlackDex** [comment]
Also, check if you are able to use Safari to connect to your instance via the web-vault.

**Comment by lethunder** [comment]
Hey guys,
same issue for me. I can sign in from my android app, web direct access but not from chrome extension and desktop app. The reason is that the endpoints are deprecated.

![image](https://user-images.githubusercontent.com/6217812/208415808-8f4c8123-ee62-46e8-9085-09c278941071.png)

https://bitwarden.com/help/november-deprecation-notice/

How can we update the base image?

**Comment by BlackDex** [comment]
Did you used the `testing` tagged docker images????
Those work with the latest clients!

**Comment by lethunder** [comment]
oh! i'm using using a docker image i installed from my synology server
https://registry.hub.docker.com/r/vaultwarden/server
i just did 
docker pull vaultwarden/server
to get latest image. Still not working

I don't understand what you mean by using the testing tagged docker image. Do you have any link for me please?

**Comment by lethunder** [comment]
i see what you mean. OK
let me test
Thanks

**Comment by lethunder** [comment]
It works! thanks

**Comment by Buddinski88** [comment]
> Also, check if you are able to use Safari to connect to your instance via the web-vault.

I just did. On OSX it works fine in the browser and with the native app.
If the cause is not Vaultwarden. Is it then the official app from Bitwarden, i.e. this repository https://github.com/bitwarden/mobile ?

**Comment by BlackDex** [comment]
For me, it works fine using the latest bitwarden mobile client. Since both Android and iOS are based upon the same code-base, both should not work. So, either you are not running the `testing` tagged docker container of Vaultwarden, or there is something not correctly set in the Bitwarden Mobile client on your side.

**Comment by Buddinski88** [comment]
That's really weird. I just really made sure again that I was using `testing`.

- Docker container stopped
- image deleted
- vaultwarden/server:testing
- docker container started "docker-compose up -d --force-recreate"
- iOS app deleted
- iOS app reinstalled

Unfortunately, it still does not work. Can I provide more information?
Btw. within iOS you can't set anything regarding the app except regarding the notifications.

**Comment by BlackDex** [comment]
> That's really weird. I just really made sure again that I was using `testing`.
> 
>     * Docker container stopped
> 
>     * image deleted
> 
>     * vaultwarden/server:testing
> 
>     * docker container started "docker-compose up -d --force-recreate"
> 
>     * iOS app deleted
> 
>     * iOS app reinstalled
> 
> 
> Unfortunately, it still does not work. Can I provide more information? Btw. within iOS you can't set anything regarding the app except regarding the notifications.

You can change the URL, which you **need** to do, else it will not connect to **your** environment.
See: https://bitwarden.com/help/change-client-environment/ > click on the `Mobile App` tab, follow the instructions there.

**Comment by Buddinski88** [comment]
> > That's really weird. I just really made sure again that I was using `testing`.
> > ```
> > * Docker container stopped
> > 
> > * image deleted
> > 
> > * vaultwarden/server:testing
> > 
> > * docker container started "docker-compose up -d --force-recreate"
> > 
> > * iOS app deleted
> > 
> > * iOS app reinstalled
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Unfortunately, it still does not work. Can I provide more information? Btw. within iOS you can't set anything regarding the app except regarding the notifications.
> 
> You can change the URL, which you **need** to do, else it will not connect to **your** environment. See: https://bitwarden.com/help/change-client-environment/ > click on the `Mobile App` tab, follow the instructions there.

Sorry, I have neglected this step, but of course. I have set this as before and with the other applications.

**Comment by Buddinski88** [comment]
**Solved**: I just had the idea to change the master password and make sure that there is not something "cached" by iOS? Voila I can also log on to my iPhone again. Hallelujah.

1. Login into webvault
2. Go to your user settings and change the master password
3. I'm not sure if it's relevant, but I also had the encryption key regenerated. This will "disconnect" all devices and they will have to re-register.
